frontend/src/pages/surveys/components/preferences-step/preferences-step.tsx

```diff
@@ -0,0 +1,110 @@
+import { type ControllerRenderProps } from 'react-hook-form';
+
+import { Button, Input, SurveyCategory } from '#libs/components/components.js';
+import {
+  useAppForm,
+  useCallback,
+  useFormController,
+} from '#libs/hooks/hooks.js';
+import {
+  type SurveyInputDto,
+  surveyInputValidationSchema,
+  SurveyTextareaOptions,
+} from '#packages/survey/survey.js';
+import {
+  DEFAULT_SURVEY_PAYLOAD,
+  TEXTAREA_MAX_LENGTH,
+} from '#pages/surveys/libs/constants.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  onSubmit: (options: string[]) => void;
+};
+
+const PreferencesStep: React.FC<Properties> = ({ onSubmit }) => {
+  const { watch, control, errors, isValid, handleSubmit } =
+    useAppForm<SurveyInputDto>({
+      defaultValues: DEFAULT_SURVEY_PAYLOAD,
+      validationSchema: surveyInputValidationSchema,
+    });
+  const { field: optionsField } = useFormController({
+    name: 'options',
+    control,
+  });
+
+  const options: string[] = [
+    'Get emotional support',
+    'Reduce stress or anxiety',
+    'Improve sleep quality',
+    'Get over with depression',
+    'Improve focus',
+    SurveyTextareaOptions.OTHER,
+  ];
```

Comment:
Let's move it outside of the component so it won't created every time the component rerenders

Reply:
Yeah, done